### PR TITLE
sort the mess with isGB

### PR DESCRIPTION
**Pull request by fieker**
follow up of Tommy's fix earlier, this now adresses this more generally

Also: add simplify(MPolyQuoElem)
          inv, isunit and isunit_with_inverse for those elements
(and a test for inv)

### CODE REVIEW

File: src/Rings/MPolyQuo.jl [hunk 1]
```diff
@@ -105,22 +115,48 @@ function quo(R::MPolyRing, I::MPolyIdeal)
...
 
 (Q::MPolyQuo)() = MPolyQuoElem(Q.R(), Q)
 (Q::MPolyQuo)(a::MPolyQuoElem) = a
 (Q::MPolyQuo)(a) = MPolyQuoElem(Q.R(a), Q)
 
 zero(Q::MPolyQuo) = Q(0)
-
 one(Q::MPolyQuo) = Q(1)
 
+function isunit_with_inverse(a::MPolyQuoElem)
```
Comment (thofma) [inline src/Rings/MPolyQuo.jl]:
```suggestion
function isinvertible_with_inverse(a::MPolyQuoElem)
```

File: src/Rings/MPolyQuo.jl [hunk 2]
```diff
@@ -105,22 +115,48 @@ function quo(R::MPolyRing, I::MPolyIdeal)
...
+  if 1 in H
+    @assert nrows(T) == 1
+    return true, Q(T[1, end])
+  end
+  return false, a
+end
+
+isunit(a::MPolyQuoElem) = isunit_with_inverse(a)[1]
+function inv(a::MPolyQuoElem)
+  fl, b = isunit_with_inverse(a)
```
Comment (thofma) [inline src/Rings/MPolyQuo.jl]:
```suggestion
  fl, b = isinvertible_with_inverse(a)
```

File: src/Rings/MPolyQuo.jl [hunk 3]
```diff
@@ -105,22 +115,48 @@ function quo(R::MPolyRing, I::MPolyIdeal)
...
+  J = vcat(J, [a.f])
+  H, T = groebner_basis_with_transform(ideal(J))
+  if 1 in H
+    @assert nrows(T) == 1
+    return true, Q(T[1, end])
+  end
+  return false, a
+end
+
+isunit(a::MPolyQuoElem) = isunit_with_inverse(a)[1]
```
Comment (thofma) [inline src/Rings/MPolyQuo.jl]:
```suggestion
isunit(a::MPolyQuoElem) = isinvertible_with_inverse(a)[1]

```

**Review by thofma — COMMENTED** [review]
We could use `isinvertible_with_inverse`, since we already have it for matrices.

**Comment by fieker** [comment]
On Tue, Feb 02, 2021 at 04:29:35AM -0800, thofma wrote:
> @thofma commented on this pull request.
> 
> We could use `isinvertible_with_inverse`, since we already have it for matrices.
> 
will do - Dan just asked also for divisibility testing, so will add as
well

**Pull request merged**